go/libkb/rpc_exim.go

```diff
@@ -1234,10 +1234,27 @@ func (u *User) ExportToUPKV2AllIncarnations() keybase1.UserPlusKeysV2AllIncarnat
 		}
 	}
 
+	// Collect the link IDs (that is, the hashes of the signature inputs) from all subchains.
+	linkIDs := []keybase1.LinkID{}
+	if u.sigChain() != nil {
+		for _, link := range u.sigChain().chainLinks {
+			// Assert that all the links are in order as they go in. We should
+			// never fail this check, but we *really* want to know about it if
+			// we do.
+			if int(link.GetSeqno()) != len(linkIDs)+1 {
+				u.G().Log.Errorf("Encountered out-of-sequence chain link %d while exporting uid %s username %s", link.GetSeqno(), u.GetUID(), u.GetName())
+				linkIDs = nil
+				break
```

Comment:
Should we change the sig to report an error here?  This might be pushing a can down the road.